src/components/TextInput/index.native.tsx

```diff
@@ -2,10 +2,11 @@ import React, {forwardRef, useEffect} from 'react';
 import {AppState, Keyboard} from 'react-native';
 import useThemeStyles from '@hooks/useThemeStyles';
 import BaseTextInput from './BaseTextInput';
-import * as baseTextInputPropTypes from './BaseTextInput/baseTextInputPropTypes';
+import type {BaseTextInputProps, BaseTextInputRef} from './BaseTextInput/types';
 
-const TextInput = forwardRef((props, ref) => {
+function TextInput(props: BaseTextInputProps, ref: BaseTextInputRef) {
```

Comment:
NAB - Question: Why aren't we destructuring `props` here, but we do in other files like `TextInputLabel`?